src/Hosting.Services.Web/Middlewares/UserIdentity/IdentityProviders/EmailBasedUserIdentityProvider.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares
+{
+	internal class EmailBasedUserIdentityProvider : IUserIdentityProvider
+	{
+        private static int MaxBytesPerChar { get; } = 4; // UTF-8
+		public int MaxBytesInIdentity { get; } = 256 * MaxBytesPerChar; // maximum email address length plus max bytes per char
+
+        public class UserEmail
+        {
+            public string Email { get; set; } = string.Empty;
+        }
+
+		public Task<(bool success, int bytesWritten)> TryWriteBytesAsync(HttpContext context, Memory<byte> memory)
+		{
+			int bytesWritten = -1;
+            try{
+                context.Request.Body.Seek(0, SeekOrigin.Begin);
+                TextReader reader = new StreamReader(context.Request.Body);
+                UserEmail? email = JsonSerializer.Deserialize<UserEmail>(reader.ReadToEnd());
```

Comment:
`.DeserializeAsync` would be more efficient. And it doesn't be too difficult to add as the class is already `async`.

```suggestion
UserEmail? email = await JsonSerializer.DeserializeAsync<UserEmail>(reader);
```